### PR TITLE
Switch from Maven to Gradle

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by yschimke** [comment]
I've found the kotlin DSL for gradle build files to be a huge win.  Slightly glitchy when you edit the build files, but getting better fast.  But more importantly I don't find myself copying and pasting gradle snippets from other projects.  The type safety of the build files and IDE completion is a massive win.

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,114 @@
+buildscript {
+  ext.versions = [
```
Comment (yschimke) [inline build.gradle]: For comparison with the Kotlin DSL

https://github.com/yschimke/okurl/blob/master/buildSrc/src/main/kotlin/Deps.kt

**Comment by Egorand** [comment]
AFAIK the biggest issue with Kotlin DSL is performance, I heard people reporting it being extremely slow. The sole fact that it spawns multiple Gradle daemons to compile different parts of the config is gross. [KotlinPoet uses KotlinDSL](https://github.com/square/kotlinpoet/blob/master/build.gradle.kts), but it's a tiny single-module project. As for OkHttp, it might be worth the experiment, but we might have to revert to Groovy if we notice IDEs becoming very slow when working with OkHttp.

**Comment by JakeWharton** [comment]
I also find its design to be bad. They tried to make it look kinda like the old Groovy instead of just designing nice, new APIs.

**Comment by yschimke** [comment]
I'm happy just to get gradle to be fair, has the open source developer mindshare right now and seems to be really pushing the envelope to stay competitive with the blaze-clones.

**Comment by swankjesse** [comment]
A couple things left for follow-up:

 * [ ] Error Prone
 * [ ] Maven publishing, including snapshot publishing from Travis

**Comment by swankjesse** [comment]
# 🐘

**Comment by yschimke** [comment]
For caching wins https://docs.travis-ci.com/user/languages/java/#caching

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**